src/main/java/com/ibm/cloud/sdk/core/http/HttpClientSingleton.java

```diff
@@ -263,6 +264,30 @@ protected SSLSocket configureSocket(SSLSocket socket) throws IOException {
     }
   }
 
+  /**
+   * Sets a new list of interceptors for the specified {@link OkHttpClient} instance by removing the specified
+   * interceptor and returns a new instance with the interceptors configured as requested.
+   *
+   * @param client the {@link OkHttpClient} instance to set the proxy authenticator on
+   * @param interceptorToRemove the class name of the interceptor to remove
+   * @return the new {@link OkHttpClient} instance with the new list of interceptors
+   */
+  private OkHttpClient reconfigureClientInterceptors(OkHttpClient client, String interceptorToRemove) {
+    OkHttpClient.Builder builder = client.newBuilder();
+
+    if (!builder.interceptors().isEmpty()) {
+      for (int i = 0; i < builder.interceptors().size(); i++) {
```

Comment:
I think you need to use an iterator to walk through the list rather than an integer index because you are potentially removing an item from the list as you are walking the list.   Iterator-based traversals allow you to remove an item safely.